### PR TITLE
CYB-465 Test of checking token emission depending of activated_share

### PR DESCRIPTION
**Pull request by sadaocasino**

### CODE REVIEW

File: tests/eosio.system_tests.cpp [hunk 1]
```diff
@@ -1186,6 +1186,57 @@ double get_continuous_rate(double emission_rate) {
...
+    // Second case with 0.33 <= activated_share <= 0.66 (-> 0.1 <= emission <= 0.2)
+    produce_block(fc::days(365));
+    BOOST_REQUIRE_EQUAL( success(), push_action( "defproducer1", N(claimrewards), mvo()("owner", "defproducer1") ) );
+
+    initial_global_state = get_global_state();
+    initial_supply = get_token_supply();
+    emission_rate = get_target_emission_per_year(1.0 * initial_global_state["active_stake"].as<int64_t>() / initial_supply.get_amount());
+    BOOST_REQUIRE ( 0.66 >= 1.0 * initial_global_state["active_stake"].as<int64_t>() / initial_supply.get_amount() &&
+                    0.33 <= 1.0 * initial_global_state["active_stake"].as<int64_t>() / initial_supply.get_amount());
+    BOOST_REQUIRE(0.1 <= emission_rate && 0.2 >= emission_rate);
```
Comment (justefg) [inline tests/eosio.system_tests.cpp]: check that
`new_supply = supply * (1 + emission_rate)`

Comment (sadaocasino) [reply tests/eosio.system_tests.cpp]: Added check after year

File: tests/eosio.system_tests.cpp [hunk 2]
```diff
@@ -1186,6 +1186,71 @@ double get_continuous_rate(double emission_rate) {
...
+    initial_supply = get_token_supply();
+    BOOST_TEST_REQUIRE(1.0 == initial_supply_after / initial_supply.get_amount());
+
+    emission_rate = get_target_emission_per_year(1.0 * initial_global_state["active_stake"].as<int64_t>() / initial_supply.get_amount());
+    BOOST_REQUIRE ( 0.66 >= 1.0 * initial_global_state["active_stake"].as<int64_t>() / initial_supply.get_amount() &&
+                    0.33 <= 1.0 * initial_global_state["active_stake"].as<int64_t>() / initial_supply.get_amount());
+    BOOST_REQUIRE(0.1 <= emission_rate && 0.2 >= emission_rate);
+
+    // Third case with 0.33 <= activated_share <= 0.66 (-> 0.1 <= emission <= 0.2)
+    years = 2;
```
Comment (justefg) [inline tests/eosio.system_tests.cpp]: remove it

File: tests/eosio.system_tests.cpp [hunk 3]
```diff
@@ -1186,6 +1186,71 @@ double get_continuous_rate(double emission_rate) {
...
+    initial_supply = get_token_supply();
+
+    emission_rate = get_target_emission_per_year(1.0 * initial_global_state["active_stake"].as<int64_t>() / initial_supply.get_amount());
+    BOOST_REQUIRE ( 0.66 >= 1.0 * initial_global_state["active_stake"].as<int64_t>() / initial_supply.get_amount() &&
+                    0.33 <= 1.0 * initial_global_state["active_stake"].as<int64_t>() / initial_supply.get_amount());
+    BOOST_REQUIRE(0.1 <= emission_rate && 0.2 >= emission_rate);
+
+    // Fourth case with 0.33 >= activated_share (-> emission = 0.2)
+    years = 7;
+    for(auto i = 0; i < years * 365; ++i) {
```
Comment (justefg) [inline tests/eosio.system_tests.cpp]: remove it

Comment (sadaocasino) [reply tests/eosio.system_tests.cpp]: removed

**Pull request merged**